echion/threads.h

```diff
@@ -327,11 +327,16 @@ void ThreadInfo::sample(int64_t iid, PyThreadState *tstate, microsecond_t delta)
         microsecond_t previous_cpu_time = cpu_time;
         update_cpu_time();
 
-        if (!is_running())
-            // If the thread is not running, then we skip it.
+        bool running = is_running();
+        if (!running && ignore_non_running_threads)
+        {
             return;
+        }
 
-        delta = cpu_time - previous_cpu_time;
+        if (running)
+        {
+            delta = cpu_time - previous_cpu_time;
```

Comment:
I think this will end up attributing some CPU time to a non-running thread (that is, to idle stacks). Perhaps we should set the `delta` explicitly to `0` if we're not ignoring running threads in this branch.

Reply:
Updated code to explicitly set it to 0